analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,394 @@
+neutral_cytobands <- cytoband_status_count %>%
+  filter(dominant_status %in% c("unstable", "neutral")) %>%
+  dplyr::pull("cytoband")
+
+# Filter the annotated CN data to include only these cytobands
+neutral_status_cytoband_df <- consensus_seg_cytoband_status_df %>%
+  filter(cytoband %in% neutral_cytobands)
+
+# Join gene-level data
+combined_neutral_status_cytoband_df <- neutral_status_cytoband_df %>%
+  left_join(
+    consensus_seg_autosomes_df,
+    by = c("Kids_First_Biospecimen_ID" = "biospecimen_id", "cytoband")
+  )
+```
+
+```{r}
+if (!(running_in_ci)) {
+  # Now count the gene-level calls (for each status call) and define
+  # the gene as that status if more than 50% of the total counts are
+  # for that particular status
+  gene_status_count <- status_majority_caller(
+    combined_neutral_status_cytoband_df,
+    "gene_symbol",
+    status_threshold
+  )
+
+  # Display table
+  gene_status_count
+}
+```
+
+## Plot calls
+
+Plot the final dominant status call on the x-axis and the fraction of each status on the y-axis.
+
+### Plot chromosome arm status calls
+
+```{r}
+# Run `plot_dominant_status_calls` function for the chromosome arm calls
+# plot_dominant_status_calls(
+#   arm_status_count
+# )
+```
+
+### Plot cytoband status calls
+
+```{r}
+# Run `plot_dominant_status_calls` function for the cytoband calls
+# plot_dominant_status_calls(
+#   cytoband_status_count
+# )
+```
+
+### Plot gene status calls
+
+```{r}
+# if (!(running_in_ci)) {
+#   # Run `plot_dominant_status_calls` function for the gene-level calls
+#   plot_dominant_status_calls(gene_status_count)
+# }
+```
+
+## Combine arm, cytoband, and gene-level status data
+
+```{r}
+# For each of the datasets we're joining, we'll only keep these columns:
+cols_to_keep <- c("Kids_First_Biospecimen_ID", "dominant_status")
+```
+
+Combine into one long data frame
+
+```{r}
+if (!(running_in_ci)) {
+  # Combine the arm, cytoband, and gene status count data
+  final_df <- bind_rows(
+    arm = select(arm_status_count,
+      cols_to_keep,
+      region = chromosome_arm
+    ),
+    cytoband = select(cytoband_status_count,
+      cols_to_keep,
+      region = cytoband
+    ),
+    gene = select(gene_status_count,
+      cols_to_keep,
+      region = gene_symbol
+    ),
+    .id = "region_type"
+  ) %>%
+    # Reorder columns more sensibly
+    select(
+      Kids_First_Biospecimen_ID,
+      status = dominant_status,
+      region,
+      region_type
+    )
+} else {
+  # Combine the arm and cytoband status count data
+  final_df <- bind_rows(
+    arm = select(arm_status_count,
+      cols_to_keep,
+      region = chromosome_arm
+    ),
+    cytoband = select(cytoband_status_count,
+      cols_to_keep,
+      region = cytoband
+    ),
+    .id = "region_type"
+  ) %>%
+    # Reorder columns more sensibly
+    select(
+      Kids_First_Biospecimen_ID,
+      status = dominant_status,
+      region,
+      region_type
+    )
+}
+
+# Print out preview
+final_df
```

Comment:
I think this final output will include neutral calls (and possibly unstable and uncallable) for cytobands and arms, but not for genes (neutral calls were not included in `consensus_seg_annotated_cn_autosomes`). We probably don't want to include neutral calls here either, though I am not sure how we should treat unstable or uncallable regions.

Reply:
In the last commit, I filtered out the neutral calls but left the unstable and uncallable calls until we decide how we should handle them.